test/integration/firestore.spec.ts

```diff
@@ -113,6 +113,46 @@ describe('admin.firestore', () => {
     expect(typeof admin.firestore.WriteResult).to.be.not.undefined;
   });
 
+  it('admin.firestore.GrpcStatus type is defined', () => {
+    expect(typeof admin.firestore.GrpcStatus).to.be.not.undefined;
+  });
+
+  it('supports operations with custom type converters', () => {
+    const converter: admin.firestore.FirestoreDataConverter<City> = {
+      toFirestore: (city: City) => {
+        return {
+          name: city.localId,
+          population: city.people,
+        };
+      },
+      fromFirestore: (snap: admin.firestore.QueryDocumentSnapshot) => {
+        return {
+          localId: snap.data().name,
+          people: snap.data().population,
+        };
+      }
+    };
+
+    const expected: City = {
+      localId: 'Sunnyvale',
+      people: 153185,
+    };
+    const refWithConverter: admin.firestore.DocumentReference<City> = admin.firestore()
+      .collection('cities')
+      .doc()
+      .withConverter(converter);
+    return refWithConverter.set(expected)
+      .then(() => {
+        return refWithConverter.get();
+      })
+      .then((snapshot: admin.firestore.DocumentSnapshot<City>) => {
+        const city: City = snapshot.data()!;
+        expect(city).to.deep.equal(expected);
```

Comment:
FWIW, since you are using converters, you should be able to do an `instanceof` check here. You would also have to change `City` to a class.

Reply:
Done